units/patch_intent.go

```diff
@@ -511,19 +511,41 @@ func (j *patchIntentProcessor) createGitHubMergeSubscription(ctx context.Context
 	catcher.Wrap(patchSub.Upsert(), "inserting patch subscription for GitHub merge queue")
 	buildSub := event.NewExpiringBuildOutcomeSubscriptionByVersion(j.PatchID.Hex(), ghSub)
 	catcher.Wrap(buildSub.Upsert(), "inserting build subscription for GitHub merge queue")
+
 	input := thirdparty.SendGithubStatusInput{
 		VersionId: j.PatchID.Hex(),
 		Owner:     p.GithubMergeData.Org,
 		Repo:      p.GithubMergeData.Repo,
 		Ref:       p.GithubMergeData.HeadSHA,
 		Desc:      "patch created",
 		Caller:    j.Name,
-		Context:   "evergreen",
 	}
-	err := thirdparty.SendPendingStatusToGithub(ctx, input, j.env.Settings().Ui.Url)
-	if err != nil {
-		catcher.Wrap(err, "failed to send patch status to GitHub")
+
+	rules, err := thirdparty.GetEvergreenBranchProtectionRules(ctx, "", p.GithubMergeData.Org, p.GithubMergeData.Repo, p.GithubMergeData.BaseBranch)
+	// We might have permission to send statuses but not to get branch
+	// protection rules, so log the error, but don't return it.
+	grip.Error(message.WrapError(err, message.Fields{
+		"job":      j.ID(),
+		"job_type": j.Type,
+		"message":  "failed to get branch protection rules",
+		"org":      p.GithubMergeData.Org,
+		"repo":     p.GithubMergeData.Repo,
+		"branch":   p.GithubMergeData.BaseBranch,
+	}))
+	if len(rules) == 0 {
```

Comment:
Can you please add a comment here explaining that this is meant to avoid a regression in case we don't have permission to get the rules?